www-src/app/replicator/fromDevice/event_synchronizer.coffee

```diff
@@ -95,7 +97,12 @@ module.exports = class EventSynchronizer
             _deleted: true
 
         @db.put toDelete, toDelete._id, toDelete._rev, (err, res) =>
-            return callback err if err
+            if err
+                if err.status is 409
+                    @conflictsHandler.handleConflicts toDelete, (err, doc) =>
+                        return callback err if err
```

Comment:
It looks like the workflow is a bit broken here. `@conflictsHandler.handleConflicts` and `@calendarSync.deleteEvent` run concurrently. And if `handleConflicts` returns an `err`, `callback` is called twice.